plugins/core/gs-appstream.c

```diff
@@ -1048,6 +1048,32 @@ gs_appstream_search (GsPlugin *plugin,
 		}
 	}
 
+	/* If the caller has provided the custom "Endless::HasDiscoveryFeedContent"
+	 * search string (which is squashed to lower case by the search machinery),
+	 * add a search for it in the custom metadata dictionary of each component.
+	 *
+	 * FIXME: Technically we should also check ` and text()='true'` in the
+	 * XPath selector, but libxmlb doesn’t support that. So this will return
+	 * all components which specify
+	 *    <custom><value key="Endless::HasDiscoveryFeedContent">false</value></custom>
+	 * too. */
+	if (g_strv_contains (values, "endless::hasdiscoveryfeedcontent")) {
```

Comment:
It seems that previously all search terms were applied to all apps' metadata, whereas this is a more targeted approach. What component makes this query? I couldn't find it in a quick search – maybe worth a cross-reference in the commit message and/or this comment?

Reply:
The query is made in `src/gs-discovery-feed-content-provider.c`. I’ll update the commit message.